config.m4

```diff
@@ -283,15 +283,12 @@ if test "$PHP_MONGODB" != "no"; then
     AC_SUBST(MONGOC_ENABLE_SHM_COUNTERS, 0)
     AC_SUBST(MONGOC_TRACE, 1)
 
-    dnl Capture user-set environment variables for metadata handshake
-    dnl We can do this after the above scripts as they only modify STD_CFLAGS
-    AS_VAR_COPY(MONGOC_USER_SET_CFLAGS, [CFLAGS])
-    AS_VAR_COPY(MONGOC_USER_SET_LDFLAGS, [LDFLAGS])
-    AS_VAR_COPY(MONGOC_CC, [CC])
```

Comment:
I don't recall why [`AS_VAR_COPY`](https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Polymorphic-Variables.html) was used here, unless it had something to do with resolving variables within CFLAGS and LDFLAGS; however, I can't make the connection to the polymorphic variable example provided in the Autoconf manual.

In any event, I don't think it was necessary for `$CC` since that should simply be the compiler name.